pkg/controllers/provisioning/provisioner.go

```diff
@@ -236,6 +236,9 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*corev1.Pod, stat
 
 	instanceTypes := map[string][]*cloudprovider.InstanceType{}
 	for _, np := range nodePools {
+		if ctx.Err() != nil {
```

Comment:
Does this section of code take so much time that we think we need to handle this error at this level? I get that there's a trade-off between the number of times that we write this and how quickly we can respond

Reply:
We probably don't need to check here. It probably makes the most sense to just check the timeout between pods in scheduling.

Reply:
Except that this context is timed out and continues on to `cloudProvider.GetInstanceTypes`. Less that we're handling the error and more that we're silencing spurious logging.

Reply:
That doesn't completely solve it right? I think we just have to handle it generally because we can still race pass this check and fire spurious errors